zipline/assets/assets.py

```diff
@@ -240,19 +240,19 @@ def _sort_future_chains(self):
         for root_symbol in self.future_chains_cache:
             self.future_chains_cache[root_symbol].sort(key=exp_key)
 
-    def _valid_contracts(self, root_symbol, as_of_date):
+    def _valid_contracts(self, root_symbol, as_of_date, knowledge_date):
         """ Returns  a list of the currently valid futures contracts
         for a given root symbol, sorted by expiration date (the
         contracts are sorted when the AssetFinder is built).
         """
         try:
             return [c for c in self.future_chains_cache[root_symbol]
-                    if c.expiration_date and (c.expiration_date > as_of_date)
-                    and c.start_date and (c.start_date <= as_of_date)]
+                    if c.expiration_date and (as_of_date <= c.expiration_date)
```

Comment:
Good point @jfkirk, that suggests to me that actually we should keep in the ```expiration_date``` and ```start_date``` clauses.